src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/LaboratoryInvestigationResult.java

```diff
@@ -11,48 +11,79 @@
 import java.util.Optional;
 
 /**
- * Example: RSL+NV+11.9:7++:::ng/mL+HI'
+ * Details of the numerical result of a Laboratory Investigation.
+ * <pre>
+ * Example: {@code RSL+NV+11.9:7++:::ng/mL+HI'}
+ * </pre>
+ * Details of the coded result of a Laboratory Investigation.
+ * <pre>
+ * Example: {@code RSL+CV+::375211000000108:921::Bowel cancer screening programme FOB test normal (finding)'}
+ * </pre>
  */
-@Getter
-@Builder
 @AllArgsConstructor
+@Builder
+@Getter
 public class LaboratoryInvestigationResult extends Segment {
 
-    private static final String KEY = "RSL";
-    private static final String QUALIFIER = "NV";
-    public static final String KEY_QUALIFIER = KEY + PLUS_SEPARATOR + QUALIFIER;
+    public static final String KEY = "RSL";
 
     private static final int MEASUREMENT_UNIT_SECTION = 4;
     private static final int MEASUREMENT_UNIT_INDEX = 3;
     private static final int DEVIATING_RESULT_INDICATOR_INDEX = 5;
 
+    private final LaboratoryInvestigationResultType resultType;
+
     private final BigDecimal measurementValue;
     private final MeasurementValueComparator measurementValueComparator;
     private final String measurementUnit;
     private final DeviatingResultIndicator deviatingResultIndicator;
 
+    private final String code;
+    private final CodingType codeType;
+    private final String description;
+
+    @SuppressWarnings("checkstyle:MagicNumber")
     public static LaboratoryInvestigationResult fromString(String edifactString) {
-        if (!edifactString.startsWith(KEY_QUALIFIER)) {
+        if (!edifactString.startsWith(KEY)) {
             throw new IllegalArgumentException("Can't create "
-                    + LaboratoryInvestigationResult.class.getSimpleName() + " from " + edifactString);
+                + LaboratoryInvestigationResult.class.getSimpleName() + " from " + edifactString);
         }
 
         final String[] keySplit = Split.byPlus(edifactString);
 
-        final BigDecimal measurementValue = extractMeasurementValue(keySplit);
-        final String measurementValueComparator = extractMeasurementValueComparator(keySplit);
-        final String measurementUnit = extractMeasurementUnit(keySplit);
-        final String deviatingResultIndicator = extractDeviatingResultIndicator(keySplit);
+        final LaboratoryInvestigationResultType resultType =
+            LaboratoryInvestigationResultType.fromCode(keySplit[1]);
+
+        final LaboratoryInvestigationResultBuilder laboratoryInvestigationResultBuilder =
+            LaboratoryInvestigationResult.builder()
+                .resultType(resultType);
 
-        return LaboratoryInvestigationResult.builder()
-            .measurementValue(measurementValue)
-            .measurementValueComparator(StringUtils.isNotBlank(measurementValueComparator)
+        if (resultType.equals(LaboratoryInvestigationResultType.NUMERICAL_VALUE)) {
```

Comment:
What behaviour should we expect if a Screening message contains an RSL+NV segment? Or if a message contains both an RSL+NV and an RSL+CV? At the moment we seem to just use the first RSL segment, should we throw an error instead?

Reply:
From looking at the EDIFACT specs:
- there can only ever be one instance of a `RSL` segment in an EDIFACT message
- `RSL+NV` is optional, and `RSL+CV` is mandatory
- `RSL+NV` will only be in a NHS003 (Pathology) message, and `RSL+CV` will only be in a NHS004 (Screening) message

So the parser will cover all of that

Reply:
I was able to get a Screening message through our EdifactParser with only an RSL+NV:

```
{
    "fullUrl": "urn:uuid:95e2355f-59d3-4dc0-9b67-943dd8535690",
    "resource": {
      "resourceType": "Observation",
      "id": "95e2355f-59d3-4dc0-9b67-943dd8535690",
      "status": "unknown",
      "code": {
        "coding": [ {
          "system": "http://snomed.info/sct",
          "code": "368481000000103",
          "display": "Bowel cancer screening result"
        } ]
      },
      "subject": {
        "reference": "urn:uuid:609797b1-eeb0-49cd-ac3b-b8b2b47e7ba0"
      },
      "performer": [ {
        "reference": "urn:uuid:e4d29e54-918c-4f3e-833a-ee1b9db9fa30"
      } ],
      "valueQuantity": {
        "value": 0.8,
        "unit": "ng/mL"
      },
      "comment": "Normal. Sent following an initial single Negative result or following 2 Negative results after an initial Weak Positive",
      "specimen": {
        "reference": "urn:uuid:4cd39dd6-ffac-4f39-9b5d-fe45d3e0814d"
      }
    }
  }
```

Reply:
That's an invalid EDIFACT message then, and goes against the spec. We shouldn't be validating anything like that in our adaptor code, we just assume the EDIFACT message is correct when it's sent to our MESH mailbox

Reply:
Alright then, just wanted to check since I was thinking about it when making the test data. Thanks!